### ISSUE TITLE
Notebook validation failed. 

### ISSUE THREAD

**Issue by melund** [snakemake/snakemake#1032]
I was working on fixing conda environment support for windows. 

In the process I stumbled on a bug in "jupyter notebook" tests.  

I get a validation error from the temporary notebooks where snakemake has added a preamble using nbformat. 

The error is triggered when snakemake runs the notebooks, but also if I just open the notebook in jupyter. 

Has anyone see something similar?

```
Notebook validation failed: Additional properties are not allowed ('id' was unexpected):
{
 "metadata": {
  "tags": [
   "snakemake-job-properties"
  ],
  "trusted": true
 },
 "id": "5d9f3b5a",
 "cell_type": "code",
 "source": "\n######## snakemake preamble start 

```

**Comment by johanneskoester** [comment]
Yes, I have seen that as well. This somehow seems to be an incompatibility between nbformat and jupyter. Luckily it does not seem to be a critical error, but of course any hint on how to solve this would be nice.

**Comment by Christian-Heyer** [comment]
Running into the same issue. A fix for now is to pin `nbformat<5.1` in the snakemake enviroment. 

It seems nbformat added cell id tags in 5.1.0 which isn't playing nice with jupyter and nbconvert.

**Comment by khughitt** [comment]
> As an aside: is it possible to run notebooks through snakemake without also creating a separate output?

I was also wondering about this.. Since the "output" for notebooks is usually the notebook itself, it would be nice to be able to drop the ~`output: test.txt` line.

Alternatively, it could be useful to support specifying an arbitrary ".html" output path, and having snakemake automatically call `nbconvert` and attempt to generate an HTML output for a notebook. Just a thought..

**Comment by johanneskoester** [comment]
The id issue is finally fixed now. It was suprisingly difficult to find the right solution, as the behavior seems a bit undocumented. And chatgpt hallucinates heavily about this :-/.
For documentation:
nbformat adds the id field, but it it not part of the notebook v4 schema. During execution, depending on the jupyter or papermill version and how strictly the notebook is validated this can fail. I now manually remove the id field after cell creation with nbformat: #3767.
